packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts

```diff
@@ -98,22 +100,15 @@ export const attachSingleEntityContext = async (
 };
 
 export const attachMultiEntityContext = async (
-  req: Request<
-    { hierarchyName: string },
-    any,
-    { entities: string[] | undefined },
-    { filter?: string }
-  > & {
+  req: Request<{ hierarchyName: string }, any, { entities: string[] }, { filter?: string }> & {
```

Comment:
`entities` can technically be undefined, but not after the validate step which is the first line of this function.

This is a classic typing + validation question, where we can receive a request of any shape, but after validation it is a well defined shape. 

My thinking on this is to move the validation out as early as possible, so the route handler has typings of the well defined shape. We should be able to pass a schema into the server-boilerplate use/get/post etc methods as an optional param which simply calls `await ajvValidate(schema, req.body);`. And gradually these schemas and typings can be moved to @tupaia/types. The schemas are auto generated from the request types, and the types can be reused in api-client as well. @rohan-bes thoughts? 

I didn't in this case as I thought it would be going too far for a _tidy as you go_ in a PR.